src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java

```diff
@@ -26,18 +30,71 @@
  */
 @WebServlet("/lecture")
 public class LectureServlet extends HttpServlet {
+
+  /* Used to create Entity and its fields */
+  private static final String LECTURE = "Lecture";
+  private static final String LECTURE_NAME = "lectureName";
+  private static final String VIDEO_URL = "videoUrl";
+  private static final String VIDEO_ID = "videoId";
+
+  /* Name of input field used for lecture name in lecture selection page. */
+  private static final String NAME_INPUT = "name-input";
+  /* Name of input field used for lecture video link in lecture selection page. */
+  private static final String VIDEO_INPUT = "video-input";
+  /* Default value if new lecture inputs are empty. */
+  private static final String DEFAULT_VALUE = "";
+
   @Override
   public void init() throws ServletException {
     // TODO: Implement LectureServlet.
   }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.sendError(500, "Not Implemented");
+    String lectureName = getParameter(request, NAME_INPUT, DEFAULT_VALUE);
+    String videoUrl = getParameter(request, VIDEO_INPUT, DEFAULT_VALUE);
+    String videoId = getVideoId(videoUrl);
+
+    // Creates Entity and stores in database
+    Entity lectureEntity = new Entity(LECTURE);
+    lectureEntity.setProperty(LECTURE_NAME, lectureName);
+    lectureEntity.setProperty(VIDEO_URL, videoUrl);
+    lectureEntity.setProperty(VIDEO_ID, videoId);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(lectureEntity);
+
+    // TODO: Add redirect link to lecture site.
   }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.sendError(500, "Not Implemented");
   }
+
+  /**
+   * Returns value with {@code name} from the {@code request} form. 
+   * If the {@code name} cannot be found, return {@code defaultValue}.
+   * @param request Form sent by client
+   * @param name {@code <input>} to read content of
+   */
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+    String value = request.getParameter(name);
+    if (value == null) {
+      return defaultValue;
+    }
+    return value;
+  }
+  
+  /** Returns YouTube video ID for a given {@code videoUrl}. */
+  private String getVideoId(String videoUrl) {
+    String pattern = "(?<=watch\\?v=|/videos/|embed\\/|youtu.be\\/|\\/v\\/|\\/e\\/|watch\\?v%3D|watch\\?feature=player_embedded&v=|%2Fvideos%2F|embed%\u200C\u200B2F|youtu.be%2F|%2Fv%2F)[^#\\&\\?\\n]*";
+    Pattern compiledPattern = Pattern.compile(pattern);
+    Matcher matcher = compiledPattern.matcher(videoUrl);
+    if (matcher.find()) {
+      return matcher.group();
```

Comment:
How hard would it be to send a request to the url to see if there's actually a video there (or if it 404s)

Reply:
Added as a todo